### PR TITLE
Add grouped message id tracker.

### PR DESCRIPTION
**Pull request by boaks**
Use a group approach to save resources.
This not intended to merge, it's just a first idea to get rid of the "HashMap" with all recently used MID/timestamps.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

**Comment by boaks** [comment]
Add `MID_PROVIDER_GROUPS` to `NetworkConfig`.

File: californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfigDefaults.java [hunk 1]
```diff
@@ -85,6 +85,7 @@ public static void setDefaults(final NetworkConfig config) {
 		config.setFloat(NetworkConfig.Keys.PROBING_RATE, 1f);
 
 		config.setBoolean(NetworkConfig.Keys.USE_RANDOM_MID_START, true);
+		config.setInt(NetworkConfig.Keys.MID_PROVIDER_GROUPS, 16);
```
Comment (sophokles73) [inline californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfigDefaults.java]: can we define a constant for the default value? makes lookup using JavaDocs much, much easier for users.

Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfigDefaults.java]: Sure.

Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfigDefaults.java]: Done.

File: californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfigDefaults.java [hunk 2]
```diff
@@ -55,6 +56,14 @@
 	 */
 	public static final int DEFAULT_BLOCKWISE_STATUS_LIFETIME = 5 * 60 * 1000; // 5 mins
 
+	/**
+	 * The default number of MID groups.
+	 * <p>
+	 * The default value is 16.
+	 * @see GroupedMessageIdTracker
+	 */
+	public static final int DEFAULT_MID_PROVIDER_GROUPS = 16;
```
Comment (sophokles73) [inline californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfigDefaults.java]: :+1:

**Pull request merged**